### ISSUE TITLE
question about deep supervision

### ISSUE THREAD

**Issue by liyunsheng13** [google-research/deeplab2#113]
Hi, thanks for the good work. I'm curious about the deep supervision used to train kmeans mentioned in the paper. What is the formulation and where is the code for it?

**Comment by yucornetto** [comment]
Hi, thanks for your interest in our work, regarding your question:

For k-means cross-attention, please see k-means [assignment](https://github.com/google-research/deeplab2/blob/main/model/layers/dual_path_transformer.py#L405) and the [update](https://github.com/google-research/deeplab2/blob/main/model/layers/dual_path_transformer.py#L425) step.

In the assignment step, an _auxiliary_result_dict_ is generated and appended to _auxiliary_outputs_ , which is finally used for loss computing in [MaX-DeepLab's loss function](https://github.com/google-research/deeplab2/blob/main/model/loss/max_deeplab_loss.py#L803) (we add a few changes to support auxiliary outputs in original MaX-DeepLab's loss).

**Comment by aquariusjay** [comment]
Closing the issue, as there is no activity for a while.
We hope your issue has been resolved.
If not, please feel free to open a new one.
